testutil/validatormock/synccomm.go

```diff
@@ -129,31 +139,64 @@ func (s *SyncCommMember) PrepareEpoch(ctx context.Context) error {
 // PrepareSlot prepares selection proofs at the start of a slot.
 func (s *SyncCommMember) PrepareSlot(ctx context.Context, slot eth2p0.Slot) error {
 	wait(ctx, s.dutiesOK)
-	selections, err := prepareSyncContributions(ctx, s.eth2Cl, s.signFunc, s.vals, s.duties, slot)
+
+	selections, err := prepareSyncSelections(ctx, s.eth2Cl, s.signFunc, s.vals, s.subcomms, slot)
 	if err != nil {
 		return err
 	}
+
 	s.setSelections(slot, selections)
 
 	return nil
 }
 
-// Message submits Sync committee messages at desired i.e., 1/3rd into the slot.
+// Message submits sync committee messages at 1/3rd into the slot.
 func (s *SyncCommMember) Message(ctx context.Context, slot eth2p0.Slot) error {
 	wait(ctx, s.dutiesOK)
 	return submitSyncMessage(ctx, s.eth2Cl, slot, s.signFunc, s.duties)
 }
 
-// Aggregate submits Sync committee messages at desired i.e., 2/3rd into the slot.
+// Aggregate submits SignedContributionAndProof at 2/3rd into the slot. It does sync committee aggregations.
+// It blocks until sync committee selections are ready for this slot.
 func (s *SyncCommMember) Aggregate(ctx context.Context, slot eth2p0.Slot) error {
-	wait(ctx, s.getSelectionsOK(slot))
+	wait(ctx, s.dutiesOK, s.getSelectionsOK(slot))
+	// TODO(xenowits): Add aggregate function.
+
 	return nil
 }
 
-func prepareSyncContributions(context.Context, eth2wrap.Client, SignFunc,
-	validators, syncDuties, eth2p0.Slot,
-) (syncSelections, error) {
-	return nil, nil
+// prepareSubcommittees returns the assignment of validators to sync subcommittees. It assumes that all validators are included in all sync subnets (subcommittees).
+func prepareSubcommittees(ctx context.Context, eth2Cl eth2wrap.Client, duties syncDuties) (subCommittees, error) {
+	spec, err := eth2Cl.Spec(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		subnetCount = spec["SYNC_COMMITTEE_SUBNET_COUNT"].(uint64)
+		subcomms    = make(subCommittees)
+	)
+	for _, duty := range duties {
+		for i := uint64(0); i < subnetCount; i++ {
+			subcomms[duty.ValidatorIndex] = append(subcomms[duty.ValidatorIndex], eth2p0.CommitteeIndex(i))
+		}
+	}
```

Comment:
```go
	var (
		subnetCount = spec["SYNC_COMMITTEE_SUBNET_COUNT"].(uint64)
		subcomms    = make(subCommittees)
		commIdx
	)
	for _, duty := range duties {
		subcomms[duty.ValidatorIndex] = append(subcomms[duty.ValidatorIndex], eth2p0.CommitteeIndex(commIdx%subnetCount))
		commIdx++
	}
```
this maybe more simple ?

Reply:
do not put logic in a `var` block, stick to simple initialisations.